custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -0,0 +1,235 @@
+        try:
+            adherence_total_doses_taken = int(adherence_total_doses_taken)
+        except ValueError:
+            adherence_total_doses_taken = 0
+
+        not_sent = case_properties.get("event_{}".format(TREATMENT_180_EVENT)) != "sent"
+        return (
+            episode_has_outcome
+            and treatment_outcome_transitioned
+            and not_sent
+            and adherence_total_doses_taken >= 180
+            and is_valid_episode_submission(episode_case)
+        )
+
+
+class BETSDrugRefillRepeater(BaseBETSRepeater):
+    friendly_name = _("BETS - Patients: Cash transfer on subsequent drug refill (voucher case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETSDrugRefillRepeaterView
+        return reverse(BETSDrugRefillRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, voucher_case):
+        if not self.case_types_and_users_allowed(voucher_case):
+            return False
+
+        voucher_case_properties = voucher_case.dynamic_case_properties()
+        episode = get_episode_case_from_voucher(voucher_case.domain, voucher_case.case_id)
+        episode_case_properties = episode.dynamic_case_properties()
+
+        def _get_voucher_count():
+            # This is an expensive operation, so only call this function if all other conditions are true.
+            voucher_count = episode_case_properties.get('approved_voucher_count', 0)
+            if voucher_count < 2:
+                voucher_count = len(
+                    get_approved_prescription_vouchers_from_episode(episode.domain, episode.case_id)
+                )
+            return voucher_count
+
+        not_sent = voucher_case_properties.get("event_{}".format(DRUG_REFILL_EVENT)) != "sent"
+
+        return (
+            # TODO: Confirm state == "fulfilled"
+            voucher_case_properties.get("state") == "fulfilled"
+            and voucher_case_properties.get("type") == "prescription"
+            and not_sent
+            and case_properties_changed(voucher_case, ['state'])
+            and is_valid_voucher_submission(voucher_case)
+            and _get_voucher_count() >= 2
+        )
+
+
+class BETSSuccessfulTreatmentRepeater(BaseBETSRepeater):
+    friendly_name = _("BETS - Patients: Cash transfer on successful treatment completion (episode case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETSSuccessfulTreatmentRepeaterView
+        return reverse(BETSSuccessfulTreatmentRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        if not self.case_types_and_users_allowed(episode_case):
+            return False
+
+        case_properties = episode_case.dynamic_case_properties()
+        not_sent = case_properties.get("event_{}".format(SUCCESSFUL_TREATMENT_EVENT)) != "sent"
```

Comment:
just to be safe, I think we should also add case_properties.get(ENROLLED_IN_PRIVATE) == 'true' here too